src/state/opamState.ml

```diff
@@ -1265,6 +1265,11 @@ let string_of_cnf string_of_atom cnf =
 let string_of_conjunction string_of_atom c =
   Printf.sprintf "%s" (OpamStd.List.concat_map " , " string_of_atom (List.rev c))
 
+let string_of_stringlist l =
```

Comment:
I've moved the 3 printing helpers inside the function. the function ```string_of_stringlist``` cleans up the input (it removes all \n) and simply concatenates the list of strings into one string.